sos/plugins/watchdog.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (C) 2018 Red Hat, Inc., Reid Wahl <nwahl@redhat.com>
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+from sos.plugins import Plugin, RedHatPlugin
+
+import re
+
+
+class Watchdog(Plugin, RedHatPlugin):
+    '''Watchdog information.'''
+    plugin_name = 'watchdog'
+    profiles = ('system',)
+    packages = ('watchdog',)
+    option_list = [
+        ('conf_file', 'watchdog config file', 'fast', '/etc/watchdog.conf'),
+    ]
+
+    def get_log_dir(self):
+        '''Get watchdog log directory.
+
+        Get watchdog log directory path configured in ``conf_file``.
+
+        :returns: The watchdog log directory path.
+        :returntype: str.
+        :raises: IOError if ``conf_file`` is not readable.
+        '''
+        log_dir = None
+        log_dir_pattern = re.compile(r'log-dir\s*=\s*(.*)')
+
+        with open(self.get_option('conf_file'), 'r') as conf_file:
+            for line in conf_file:
+                line = line.split('#')[0].strip()
+
+                try:
+                    match = log_dir_pattern.match(line)
+                    log_dir = match.group(1)
+                except AttributeError:
+                    pass
+
+        return log_dir
+
+    def setup(self):
+        '''Collect watchdog information.
+
+        Collect configuration files, custom executables for test-binary
+        and repair-binary, and stdout/stderr logs.
+        '''
+        watchdog_conf = self.get_option('conf_file')
+        log_dir = '/var/log/watchdog'
+
+        # Get service configuration and sysconfig files
+        self.add_copy_spec([
+            watchdog_conf,
+            '/etc/sysconfig/watchdog',
+        ])
+
+        # Get custom executables
+        self.add_copy_spec([
+            '/etc/watchdog.d',
+            '/usr/libexec/watchdog/scripts',
+        ])
+
+        # Get logs
+        try:
+            res = self.get_log_dir()
+            if res:
+                log_dir = res
+        except IOError as ex:
+            self._log_warn("Could not parse %s: %s" % (watchdog_conf, ex))
```

Comment:
"Could not read" may be a better message here, since it matches the type of exception you are catching ("Could not parse" could be seen to suggest a syntactic problem with the file content).